File: app/models/qdm/medication_request.rb
module QDM
  # app/models/qdm/medication_request.rb
  class MedicationRequest < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :identifier
    field :status
    field :statusReason
    field :intent
    field :category
    field :priority
    field :doNotPerform
    field :reported
    field :medication
    field :subject
    field :encounter
    field :supportingInformation
    field :authoredOn
    field :requester
    field :performer
    field :performerType
    field :recorder
    field :reasonQDM::Code
    field :reasonReference
    field :instantiatesCanonical
    field :instantiatesUri
    field :basedOn
    field :groupIdentifier
    field :courseOfTherapyType
    field :insurance
    field :note
    field :dosageInstruction
    field :dispenseRequest
    field :substitution
    field :priorPrescription
    field :detectedIssue
    field :eventHistory
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/MedicationRequest.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const MedicationRequestSchema = DataElementSchema({
  identifier: Any,
  status: Any,
  statusReason: Any,
  intent: Any,
  category: Any,
  priority: Any,
  doNotPerform: Any,
  reported: Any,
  medication: Any,
  subject: Any,
  encounter: Any,
  supportingInformation: Any,
  authoredOn: Any,
  requester: Any,
  performer: Any,
  performerType: Any,
  recorder: Any,
  reasonCode: Any,
  reasonReference: Any,
  instantiatesCanonical: Any,
  instantiatesUri: Any,
  basedOn: Any,
  groupIdentifier: Any,
  courseOfTherapyType: Any,
  insurance: Any,
  note: Any,
  dosageInstruction: Any,
  dispenseRequest: Any,
  substitution: Any,
  priorPrescription: Any,
  detectedIssue: Any,
  eventHistory: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'MedicationRequest' },

});

module.exports.MedicationRequestSchema = MedicationRequestSchema;
module.exports.MedicationRequest = mongoose.model('MedicationRequest', MedicationRequestSchema);
